### PR TITLE
Rely on `__cuda_array_interface__` for tensor conversions

### PR DESCRIPTION
**Pull request by emcastillo**
The bugs were already fixed in PyTorch, so do the conversions directly avoiding hacks.

### CODE REVIEW

File: chainer_pytorch_migration/tensor.py [hunk 1]
```diff
@@ -59,7 +46,7 @@ def astensor(array):
         view are gone.
 
     Note:
-        If the array has negative strides, a copy is made
+        aaIf the array has negative strides, a copy is made
```
Comment (kmaehashi) [inline chainer_pytorch_migration/tensor.py]:
```suggestion
        If the array has negative strides, a copy is made
```

File: chainer_pytorch_migration/tensor.py [hunk 2]
```diff
@@ -27,20 +27,7 @@ def asarray(tensor):
...
-                ),
-                tensor.storage_offset() * itemsize,
-            )
-            return cupy.ndarray(
-                tuple(tensor.shape),
-                dtype=to_numpy_dtype(tensor.dtype),
-                memptr=memptr,
-                strides=tuple(s * itemsize for s in tensor.stride()),
-            )
+            return cupy.array(tensor.detach(), copy=False)
```
Comment (emcastillo) [inline chainer_pytorch_migration/tensor.py:30]: This is not correct, detach creates a copy ...
